tools/model-compiler/src/main/java/io/spine/tools/compiler/rejection/RejectionWriter.java

```diff
@@ -121,62 +124,77 @@ public void write() {
-            String parameterName = FieldName.of(field.getKey())
-                                            .javaCase();
-            builder.addParameter(parameterType, parameterName);
-        }
-
-        return builder.addStatement(superStatement())
-                      .build();
-    }
-
-    private String superStatement() {
-        StringBuilder superStatement = new StringBuilder("super(");
-        superStatement.append(declaration.getOuterJavaClass())
-                      .append('.')
-                      .append(declaration.getSimpleJavaClassName())
-                      .append(".newBuilder()");
-
-        for (Map.Entry<String, FieldType> field : fieldDeclarations().entrySet()) {
-            FieldName fieldName = FieldName.of(field.getKey());
-            superStatement.append('.')
-                          .append(field.getValue()
-                                       .getSetterPrefix())
-                          .append(fieldName.toCamelCase())
-                          .append('(')
-                          .append(fieldName.javaCase())
-                          .append(')');
-        }
-        superStatement.append(".build())");
-
-        return superStatement.toString();
+        ParameterSpec builderParameter = builder.asParameter();
+        CodeBlock buildRejectionMessage = builder.buildRejectionMessage();
+        return constructorBuilder()
+                .addJavadoc(constructorJavadoc(builderParameter))
+                .addModifiers(PRIVATE)
+                .addParameter(builderParameter)
+                .addStatement("super($L)", buildRejectionMessage.toString())
+                .build();
     }
 
     private MethodSpec getMessageThrown() {
         String methodSignature = getMessageThrown.signature();
         log().debug("Constructing method {}", methodSignature);
 
-        TypeName returnType =
-                ClassName.get(declaration.getJavaPackage()
-                                         .value(),
-                              declaration.getOuterJavaClass()
-                                         .value())
-                         .nestedClass(declaration.getSimpleJavaClassName()
-                                                 .value());
+        ClassName returnType = new GeneratedRejectionDeclaration(declaration).rejectionMessage();
         return MethodSpec.methodBuilder(getMessageThrown.name())
                          .addAnnotation(Override.class)
                          .addModifiers(PUBLIC)
                          .returns(returnType)
-                         .addStatement("return (" + returnType + ") super." + methodSignature)
+                         .addStatement("return ($T) super.$L", returnType, methodSignature)
                          .build();
     }
 
+    /**
+     * Generates a Javadoc content for the rejection.
+     *
+     * @return the class-level Javadoc content
+     */
+    private CodeBlock classJavadoc() {
```

Comment:
Now two classes generate code blocks for the rejection: this one and the `RejectionBuilder`. It's weird.

Reply:
How would you separate responsibilities? For me, it looks ok, that the `RejectionWriter` delegates the builder generation to the `RejectionBuilder`.